File: abc/abc148/abc148_d.cs
// https://atcoder.jp/contests/abc148/submissions/9072633

using System;
using System.Linq;

namespace CSharp
{
    class Program
    {
        static void Main()
        {
            var N = int.Parse(Console.ReadLine());
            var A = Console.ReadLine().Split(' ').Select(int.Parse).ToArray();

            if (!A.Contains(1))
            {
                Console.WriteLine("-1");
                return;
            }

            var temp = 1;
            for (int i = 0; i < N; i++)
            {
                if (A[i] == temp)
                {
                    temp++;
                }
            }
            Console.WriteLine(N - temp + 1);
        }
    }
}

File: abc/abc148/abc148_d.fs
// https://atcoder.jp/contests/abc148/submissions/12180973

let N = stdin.ReadLine() |> int
let A = stdin.ReadLine().Split() |> Array.map int

let rec solver s i =
    if i = N then N - s + 1
    elif A.[i] = s then solver (s + 1) (i + 1)
    else solver s (i + 1)

if not (A |> Array.contains 1) then -1
else solver 1 0
|> stdout.Write
